packages/sdk/src/serialization.ts

```diff
@@ -144,9 +196,27 @@ export function serializeAccountTransactionPayload(accountTransaction: Omit<Acco
  */
 export function getAccountTransactionHash(
     accountTransaction: AccountTransaction,
-    signatures: AccountTransactionSignature
+    signatures: AccountTransactionSignature,
+    givenEnergy?: Energy.Type
 ): string {
-    const serializedAccountTransaction = serializeAccountTransaction(accountTransaction, signatures);
+    let serializedAccountTransaction;
+    if (
+        accountTransaction.type === AccountTransactionType.Update ||
+        accountTransaction.type === AccountTransactionType.InitContract
+    ) {
+        if (givenEnergy === undefined) throw new Error('need to specify energy for contract transactions');
+        serializedAccountTransaction = serializeAccountTransaction(
+            accountTransaction as AccountTransaction<InitUpdateType, InitUpdatePayload>,
+            signatures,
+            givenEnergy
+        );
+    } else {
+        serializedAccountTransaction = serializeAccountTransaction(
+            accountTransaction as AccountTransaction<OtherType, OtherPayload>,
+            signatures
+        );
+    }
+
```

Comment:
```suggestion
    const serializedAccountTransaction = serializeAccountTransaction(accountTransaction, signatures);
```

Does this not still work? Otherwise it might make sense to include a completely dynamic (internal, i.e. _not_ exported) overload of it:

```
function serializeAccountTransaction(
    accountTransaction: AccountTransaction,
    signatures: AccountTransactionSignature,
    givenEnergy?: Energy.Type
): Buffer;
```

Reply:
if we didn't use the IF ELSE, there is a compiler error stating "Argument of type 'AccountTransaction<AccountTransactionType, AccountTransactionPayload>' is not assignable to parameter of type 'AccountTransaction<AccountTransactionType.DeployModule | AccountTransactionType.Transfer | AccountTransactionType.AddBaker | AccountTransactionType.RemoveBaker | ... 15 more ... | AccountTransactionType.TokenUpdate, SimpleTransferPayload | ... 6 more ... | TokenUpdatePayload>'.
  Type 'AccountTransactionType' is not assignable to ty...."